File: packages/js/src/schemas/enrollment.ts
import { z } from "zod";

import { BaseResponseSchema } from "./common.js";

export const EnrollmentInformationResponseSchema = BaseResponseSchema.extend({
  data: z.object({
    codSemestre: z.string(),
    fechaDB: z.string(),
    fecIniMatInternet: z.string(),
    fecFinMatInternet: z.string(),
    mensajeMat: z.string(),
    indMatHabilitada: z.boolean(),
    matricula: z.unknown().nullable(),
  }).passthrough(),
}).passthrough();

export type EnrollmentInformationResponse = z.infer<typeof EnrollmentInformationResponseSchema>;

File: packages/python/src/unmsm_sum_api/schemas/enrollment.py
from typing import Any

from .common import BaseResponse, FlexibleModel


class EnrollmentInformation(FlexibleModel):
    codSemestre: str
    fechaDB: str
    fecIniMatInternet: str
    fecFinMatInternet: str
    mensajeMat: str
    indMatHabilitada: bool
    matricula: Any | None


class EnrollmentInformationResponse(BaseResponse):
    data: EnrollmentInformation
